api/v1beta1/awsloadbalancercontroller_types.go

```diff
@@ -0,0 +1,206 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:validation:Enum=AWSShield;AWSWAFv1;AWSWAFv2
+type AWSAddon string
+
+const (
+	AWSAddonShield AWSAddon = "AWSShield"
+	AWSAddonWAFv1  AWSAddon = "AWSWAFv1"
+	AWSAddonWAFv2  AWSAddon = "AWSWAFv2"
+)
+
+// +kubebuilder:validation:Enum=Auto;Manual
+type SubnetTaggingPolicy string
+
+const (
+
+	// AutoSubnetTaggingPolicy enables automatic subnet tagging.
+	AutoSubnetTaggingPolicy SubnetTaggingPolicy = "Auto"
+
+	// ManualSubnetTaggingPolicy disables automatic subnet tagging.
+	ManualSubnetTaggingPolicy SubnetTaggingPolicy = "Manual"
+)
+
+// AWSLoadBalancerControllerSpec defines the desired state of AWSLoadBalancerController
+type AWSLoadBalancerControllerSpec struct {
+
+	// SubnetTagging describes how resource tagging will be done by the operator.
+	//
+	// When in "Auto", the operator will detect the subnets where the load balancers
+	// will be provisioned and have the required resource tags on them. Whereas when
+	// set to manual, this responsibility lies on the user.
+	//
+	// +kubebuilder:default:=Auto
+	// +kubebuilder:validation:Optional
+	// +optional
+	SubnetTagging SubnetTaggingPolicy `json:"subnetTagging,omitempty"`
+
+	// Default AWS Tags that will be applied to all AWS resources managed by this
+	// controller (default []).
+	//
+	// This value is required so that this controller can function as expected
+	// in parallel to openshift-router.
+	//
+	// +kubebuilder:default:={}
+	// +kubebuilder:validation:Optional
+	// +optional
+	AdditionalResourceTags map[string]string `json:"additionalResourceTags,omitempty"`
+
+	// IngressClass specifies the Ingress class which the controller will reconcile.
+	// This Ingress class will be created unless it already exists.
+	// The value will default to "alb".
+	//
+	// +kubebuilder:default:=alb
+	// +kubebuilder:validation:Optional
+	// +optional
+	IngressClass string `json:"ingressClass,omitempty"`
+
+	// Config specifies further customization options for the controller's deployment spec.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	Config *AWSLoadBalancerDeploymentConfig `json:"config,omitempty"`
+
+	// AWSAddon describes the AWS services that can be integrated with
+	// the AWS Load Balancer.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	EnabledAddons []AWSAddon `json:"enabledAddons,omitempty"` // indicates which AWS addons should be disabled.
```

Comment:
What happens when an item is removed from teh list?

Reply:
Each item in this list is 1-1 mapping to a boolean addon flag in the operand. All the addons flags are set to `false` by default and reset to `true` only for the items mentioned in the list. So, if an item is removed the corresponding flag will be set back to the default: `false`.

Reply:
Update the comment to clarify the behavior (and the why) of the enabling and disabling of the addons.